src/wfs/src/main/java/org/geoserver/wfs/xml/GML3OutputFormat.java

```diff
@@ -243,6 +243,12 @@ protected void write(
         Encoder encoder = createEncoder(configuration, ns2metas, gft);
 
         encoder.setEncoding(Charset.forName(geoServer.getSettings().getCharset()));
+        if (wfs.isVerbose() || geoServer.getSettings().isVerbose()) {
+            geoServer.getSettings().setVerbose(true);
```

Comment:
For every invocation of this output format, a configuration change is made? That does not seem right....
(also, it's not persisted, but it's sort of a bonus in this situation).

Reply:
Good spot - that was when I was debugging which set of settings need to be set.